### DISCUSSION TITLE
OctoberCMS: Multi router support

### DISCUSSION THREAD

**Discussion by samuelpatro** [getsentry/sentry-laravel#710] [Q&A]
Hey,
not sure if there is a way to overwrite or bind if there are more routers that make other routes but OctoberCMS uses 2 different routes for backend and frontend controllers. Sentry reports them as same per transaction is there a way to overwrite this?

Now it shows up in sentry like this, but they are grouped as same but under that are like 20 different routes:

![image](https://github.com/getsentry/sentry-laravel/assets/3110002/48c912a8-3b20-46f5-9bb4-b164272ee94e)

**Comment by stayallive** [comment]
It depends on how this is structured, but we could definitly do better here, I have no clue how OctoberCMS is structure and if/how we can access that second/nested router. Would be a great enhancement!

**Reply by samuelpatro** [reply]
I was thinking to check if route is one of them then map them by specific class that has info about all that generated routes. Then i can somehow change up code inside where is route mappeed to transaction name and see get correct route. 

Here is how other routes are created, they are all generated dynamically.
backend routes: https://github.com/octobercms/october/blob/1.1/modules/backend/routes.php#L28
frontend routes: https://github.com/octobercms/october/blob/1.1/modules/cms/routes.php#L25

**Comment by cleptric** [comment]
A few options to solve this

- You may use different Sentry projects and swap out the DSN based on the route.
- You update the transaction name inside `before_send_transaction` based on the current domain.
- You inject a custom middleware into the route groups and update the transaction name there, though I'm not sure OctoberCMS supports this.

**Reply by samuelpatro** [reply]
> You update the transaction name inside before_send_transaction based on the current domain.

i want to do something like this, or to change up this method (https://github.com/getsentry/sentry-laravel/blob/master/src/Sentry/Laravel/EventHandler.php#L249), but iam now not sure what methods should i overwrite to test it out, maybe even overwrite class to create new event handler totally that can check for this differencies, like they did here but its outdated https://github.com/OFFLINE-GmbH/oc-sentry-plugin

**Reply by stayallive** [reply]
The repo you mention is an exact copy of an older version of the Laravel SDK, so their event handler is actually the one we shipped a few releases ago and indeed works a little different where we used to show route names and/or controller methods. Although I don't think that would help in this case since that resolved to the same names anyway. 

As mentioned you can override the transaction name in both `before_send` (for errors) and `before_send_transaction` for performance transactions. So if you can figure out the route OctoberCMS is serving you should be able to set that correctly and end up with better transaction names. I (really) quickly browsed in the OctoberCMS code and didn't see any events or something we could hook into to pull that info ourself, but if you figure it out I'd be happy to see if there is a way we could pull the into the Laravel SDK if it's a good fit.

**Reply by samuelpatro** [reply]
@stayallive thanks, its there a way to test it locally trough sentry:test command? I tried dd() it inside it but still cotinues without return.

**Reply by stayallive** [reply]
Yeah, the test command is using some special handling and disables most user configuration. You can test by doing `\Sentry\captureException($exception)` to sent an exception. See: https://docs.sentry.io/platforms/php/guides/laravel/usage/.

**Reply by samuelpatro** [reply]
Testing this setup for both before events for now.

```
        if ($transaction->getTransaction() == '/admin/{slug?}' || $transaction->getTransaction() == '/{slug?}') {
            $transaction->setTransaction(\Request::getRequestUri());
        }
```
